torch/csrc/jit/codegen/cuda/lower_validation.cpp

```diff
@@ -88,8 +88,78 @@ void validateIr(Fusion* fusion) {
 
 namespace {
 
+// Check contiguity for all root domains associated with Misaligned Vectorize ParallelType
+void checkContiguity(
+    const std::unordered_set<IterDomain*>& domains,
+    TensorView* tv) {
+  TORCH_INTERNAL_ASSERT(tv->getMemoryType() == MemoryType::Global);
+
+  for (size_t idx = 0; idx < tv->getRootDomain().size(); ++idx) {
+    auto root = tv->getRootDomain()[idx];
+    if (domains.find(root) != domains.end()) {
+      TORCH_INTERNAL_ASSERT(
+          !root->isBroadcast(),
+          "Misaligned vectorization prohibits merging broadcast domains.",
+          "Issue found in, ",
+          tv);
+      TORCH_INTERNAL_ASSERT(
+          tv->domain()->contiguity()[idx],
+          "Cannot merge non-contiguous root domains with misaligned vectorization.",
+          "Issue found in, ",
+          tv);
+    }
+  }
+}
+
+// Check contiguity for all root domains associated with Misaligned Vectorize ParallelType
+void checkContiguity(
```

Comment:
`assertMappedConsumerContiguity`

Reply:
I don't know what property you're checking for here, so I really don't know if this is doing what you expect.

Reply:
We cannot merge any non-contiguous root domains for misaligned vectorization (MV).
This function checks if all the root domains associated with MV iter_domain are contiguous.

We apply the MV parallel type to the local TensorView, but the contiguity information is stored with the global TensorView. This function uses the PairwiseRootDomainMap to map the root domains between the TensorViews.